sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -256,64 +260,94 @@ void finalizeAndValidate(RelationName relationName,
-                                                    ExpressionAnalyzer expressionAnalyzer,
-                                                    SymbolPrinter printer,
-                                                    ExpressionAnalysisContext expressionAnalysisContext) {
+    private static void processExpressions(AnalyzedColumnDefinition columnDefinition,
+                                           ExpressionAnalyzer generatedExpressionAnalyzer,
+                                           ExpressionAnalysisContext generatedExpressionAnalysisContext,
+                                           ExpressionAnalyzer defaultExpressionAnalyzer,
+                                           ExpressionAnalysisContext defaultExpressionAnalysisContext,
+                                           SymbolPrinter printer) {
         if (columnDefinition.generatedExpression() != null) {
-            processGeneratedExpression(expressionAnalyzer, printer, columnDefinition, expressionAnalysisContext);
+            processExpression(generatedExpressionAnalyzer,
+                              columnDefinition::generatedExpression,
+                              columnDefinition::formattedGeneratedExpression,
+                              printer,
+                              columnDefinition,
+                              generatedExpressionAnalysisContext);
+        }
+        if (columnDefinition.defaultExpression() != null) {
+            processExpression(defaultExpressionAnalyzer,
+                              columnDefinition::defaultExpression,
+                              columnDefinition::formattedDefaultExpression,
+                              printer,
+                              columnDefinition,
+                              defaultExpressionAnalysisContext);
         }
         for (AnalyzedColumnDefinition child : columnDefinition.children()) {
-            processGeneratedExpressions(
+            processExpressions(
                 child,
-                expressionAnalyzer,
-                printer,
-                expressionAnalysisContext
+                generatedExpressionAnalyzer,
+                generatedExpressionAnalysisContext,
+                defaultExpressionAnalyzer,
+                defaultExpressionAnalysisContext,
+                printer
             );
         }
     }
 
-    private static void processGeneratedExpression(ExpressionAnalyzer expressionAnalyzer,
-                                                   SymbolPrinter symbolPrinter,
-                                                   AnalyzedColumnDefinition columnDefinition,
-                                                   ExpressionAnalysisContext expressionAnalysisContext) {
-        // validate expression
-        Symbol function = expressionAnalyzer.convert(columnDefinition.generatedExpression(), expressionAnalysisContext);
+    private static void processExpression(ExpressionAnalyzer expressionAnalyzer,
```

Comment:
Might want to inline this method. The creation of the `expressionSupplier` and `formattedExpressionConsumer` + the selection of the right expressionAnalyzer is about as complicated as the logic that is encapsulated here.

Reply:
Yeah, I guess you are right... 
Most of the code duplication was avoided by introducing the 
`validateAndFormatExpression()` method. I might have overdid it here. 
Thanks!